File: 4way/problem005/solution.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function(arr) {
    const numberOccurrences = {};
    
    // Contar las ocurrencias
    for (let num of arr) {
        numberOccurrences[num] = (numberOccurrences[num] || 0) + 1;
    }
    
    const occurrences = new Set();
    
    // Verificar si las frecuencias son únicas
    for (let count of Object.values(numberOccurrences)) {
        if (occurrences.has(count)) {
            return false;
        }
        occurrences.add(count);
    }
    
    return true;
};

File: 4way/problem005/solution.py
class Solution(object):
    def uniqueOccurrences(self,arr):
        # Contar las ocurrencias
        numberOccurrences = {}
        for num in arr:
            numberOccurrences[num] = numberOccurrences.get(num, 0) + 1

        # Verificar si las frecuencias son únicas
        occurrences = set()
        for count in numberOccurrences.values():
            if count in occurrences:
                return False
            occurrences.add(count)
        
        return True
